nemo/collections/asr/parts/submodules/fast_rnnt_greedy_decoding.py

```diff
@@ -0,0 +1,440 @@
+        )
+    )
+    torch.cuda.set_stream(body_stream)
+
+    yield body_stream, body_graph
+
+    cuda.cuLaunchKernel(
+        while_loop_kernel, 1, 1, 1, 1, 1, 1, 0, body_stream.cuda_stream, while_loop_args.ctypes.data, 0
+    )
+
+    cudart.cudaStreamEndCapture(body_stream.cuda_stream)
+
+    torch.cuda.set_stream(previous_stream)
+
+
+class RNNTGreedyDecodeFast:
+    def __init__(self, max_symbols: int, cuda_device, caller):
+        assert max_symbols is not None
+
+        self.symbols_added_t = torch.tensor(0, dtype=torch.int64, device=cuda_device)
+        self.max_symbols_t = torch.tensor(max_symbols, dtype=torch.int64, device=cuda_device)
+        self.max_symbols = max_symbols
+        self.not_all_blank_t = torch.tensor(True, dtype=torch.bool, device=cuda_device)
+
+        self.cuda_device = cuda_device
+
+        self.time_idx_t = torch.tensor(0, dtype=torch.int64, device=self.cuda_device)
+        self.max_out_len_t = torch.tensor(0, dtype=torch.int64, device=self.cuda_device)
+
+        self.encoder_output = None
+        self.encoder_output_length = None
+        self.f = None
+
+        # Reasonable default maximum time. 375 frames * 40ms / frame = 15 seconds
+        # This affects only the size of the CPU-pinned memory buffers
+        self.max_time = 375
+        self.batch_size = 0
+
+        self.scores_cpu = None
+        self.labels_cpu = None
+        self.graph = None
+        self.graph_exec = None
+
+        self.caller = caller
+
+    def _reinitialize(self, max_time, batch_size, encoder_output, encoder_output_length):
+        # We need to call _greedy_decode_blank_as_pad at least once
+        # before hand in order to make sure that pytorch is
+        # "initialize".
+        self.caller._greedy_decode_blank_as_pad_loop_frames(
+            encoder_output, encoder_output_length, encoder_output.device
+        )
+
+        self.max_time = max(self.max_time, max_time)
+        self.batch_size = max(self.batch_size, batch_size)
+
+        self.encoder_output = torch.zeros(
+            (self.batch_size, self.max_time, encoder_output.shape[-1]),
+            dtype=encoder_output.dtype,
+            device=encoder_output.device,
+        )
+        self.encoder_output_length = torch.zeros(
+            (self.batch_size,), dtype=encoder_output_length.dtype, device=encoder_output_length.device
+        )
+
+        self.zero_t = torch.tensor(0.0, dtype=encoder_output.dtype, device=encoder_output.device)
+        self.blank_index_t = torch.tensor(self.caller._blank_index, dtype=torch.long, device=encoder_output.device)
+
+        self.scores_cpu = torch.zeros(
+            (self.batch_size, self.max_time, self.max_symbols),
+            dtype=encoder_output.dtype,
+            device="cpu",
+            pin_memory=True,
+        )
+        self.labels_cpu = torch.zeros(
+            (self.batch_size, self.max_time, self.max_symbols), dtype=torch.int64, device="cpu", pin_memory=True
+        )
+
```

Comment:
Add comment this is start of decoding loop capture 

In general, document explicitly each step so that future read can keep track of what is going on  in case there is need to modify